### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This is a mix of:
* comment typos
* local typos
* api typos
* actual bugs
* user visible typos
* brand typos

I can split / rearrange if requested, but I'd rather not spend time if you aren't interested.

FWIW, I haven't fixed certain instances in the ChangeLog where you mention correcting a typo'd term. I'd highly encourage you *not* to include misspellings in commit messages / ChangeLogs, it's just bad. In 99% of cases, no one cares how something was previously spelled. And any cases where you've included a typo intentionally make it harder for tooling to identify unintentional typos.

### CODE REVIEW

File: frontends/php/chart5.php [hunk 1]
```diff
@@ -77,7 +77,7 @@
 	'gridbordercolor' => 'ACBBC2',
 	'nonworktimecolor' => 'EBEBEB',
 	'leftpercentilecolor' => '429E47',
-	'righttpercentilecolor' => 'E33734'
+	'rightpercentilecolor' => 'E33734'
```
Comment (jsoref) [inline frontends/php/chart5.php]: elsewhere the code used the correct spelling...

File: frontends/php/hostinventoriesoverview.php [hunk 2]
```diff
@@ -68,7 +68,7 @@
 $_REQUEST['groupby'] = getRequest('groupby', '');
 $groupFieldTitle = '';
 
-$hostinvent_wdgt = (new CWidget())->setTitle(_('Host inventory overview'));
+$hostinventory_wdgt = (new CWidget())->setTitle(_('Host inventory overview'));
```
Comment (jsoref) [inline frontends/php/hostinventoriesoverview.php]: this isn't technically a misspelling, but there's no reason not to spell it out

File: frontends/php/include/classes/api/services/CUser.php [hunk 3]
```diff
@@ -391,8 +391,8 @@ public function create($users) {
 
 			$usrgrps = zbx_objectValues($user['usrgrps'], 'usrgrpid');
 			foreach ($usrgrps as $groupid) {
-				$usersGroupdId = get_dbid('users_groups', 'id');
-				$sql = 'INSERT INTO users_groups (id,usrgrpid,userid) VALUES ('.zbx_dbstr($usersGroupdId).','.zbx_dbstr($groupid).','.zbx_dbstr($userid).')';
+				$usersGroupId = get_dbid('users_groups', 'id');
```
Comment (jsoref) [inline frontends/php/include/classes/api/services/CUser.php]: I'm not absolutely confident about this change...

File: frontends/php/include/classes/items/CHelpItems.php [hunk 4]
```diff
@@ -142,7 +142,7 @@ protected function getItems() {
 				],
 				[
 					'key' => 'proc.cpu.util[<name>,<user>,<type>,<cmdline>,<mode>,<zone>]',
-					'description' => _('Process CPU utilisation percentage. Returns float')
+					'description' => _('Process CPU utilization percentage. Returns float')
```
Comment (jsoref) [inline frontends/php/include/classes/items/CHelpItems.php]: This is an example of an en-GB -> en-US change, but note that the en-US form was already dominant.

File: frontends/php/js/class.cmessages.js [hunk 5]
```diff
@@ -505,7 +505,7 @@ var CNode = Class.create({
 			this.node.appendChild(item);
 		}
 		else if (is_string(item)) {
-			this.node.appendChild(documect.createTextNode(item));
+			this.node.appendChild(document.createTextNode(item));
```
Comment (jsoref) [inline frontends/php/js/class.cmessages.js]: this is a bug fix

File: frontends/php/js/main.js [hunk 6]
```diff
@@ -408,7 +408,7 @@ var hintBox = {
 		return box;
 	},
 
-	HintWraper: function(e, target, hintText, className, styles) {
+	HintWrapper: function(e, target, hintText, className, styles) {
```
Comment (jsoref) [inline frontends/php/js/main.js]: This is a change to a js api (the corresponding change to the html was earlier)

File: frontends/php/screenconf.php [hunk 7]
```diff
@@ -209,7 +209,7 @@
 
 		if (getRequest('form') === 'full_clone') {
 			$output = ['resourcetype', 'resourceid', 'width', 'height', 'x', 'y', 'colspan', 'rowspan', 'elements',
-				'valign', 'haligh', 'style', 'url', 'max_columns'
+				'valign', 'halign', 'style', 'url', 'max_columns'
```
Comment (jsoref) [inline frontends/php/screenconf.php]: I'm not sure what this code was doing, but it wasn't doing the right thing for `halign` before...

File: frontends/php/styles/blue-theme.css [hunk 8]
```diff
@@ -3052,7 +3052,7 @@ Text on html5 input types
    </li>
    <li>
       <div class="table-forms-td-left"><label for="DisabledInput">Disabled Input</label></div>
-      <div class="table-forms-td-right"><input type="text" placeholder="Text" id="DisabledtInput" disabled></div>
+      <div class="table-forms-td-right"><input type="text" placeholder="Text" id="DisabledInput" disabled></div>
```
Comment (jsoref) [inline frontends/php/styles/blue-theme.css]: This change could affect js/css/html.

File: frontends/php/styles/blue-theme.css [hunk 9]
```diff
@@ -3088,7 +3088,7 @@ Text on html5 input types
 
 ```html_example
 <label for="Radio"><input type="radio" id="Radio">I am a radio</label>
-<label for="RadioDisbld"><input type="radio" id="RadioDisbld" disabled>I am a disabled radio</label>
+<label for="RadioDisabled"><input type="radio" id="RadioDisabled" disabled>I am a disabled radio</label>
```
Comment (jsoref) [inline frontends/php/styles/blue-theme.css]: This is a matched pair

File: frontends/php/styles/blue-theme.css [hunk 10]
```diff
@@ -3105,7 +3105,7 @@ Text on html5 input types
 
 ```html_example
 <label for="CheckBox"><input type="checkbox" id="CheckBox">I am a checkbox</label>
-<label for="ChecBoxkDisbld"><input type="checkbox" id="CheckBoxDisbld" disabled>I am a disabled checkbox</label>
+<label for="CheckBoxDisabled"><input type="checkbox" id="CheckBoxDisabled" disabled>I am a disabled checkbox</label>
```
Comment (jsoref) [inline frontends/php/styles/blue-theme.css]: This change fixes a bug in the pair.

File: src/libs/zbxdb/db.c [hunk 11]
```diff
@@ -65,7 +65,7 @@ struct zbx_db_result
 	int		cursor;
 	DB_ROW		values;
 #elif defined(HAVE_SQLITE3)
-	int		curow;
+	int		currow;
```
Comment (jsoref) [inline src/libs/zbxdb/db.c]: I'm pretty sure this is `current row`

File: src/libs/zbxsysinfo/win32/net.c [hunk 12]
```diff
@@ -24,12 +24,12 @@
...
 /* __stdcall calling convention is used for GetIfEntry2(). In order to declare a */
 /* pointer to GetIfEntry2() we have to expand NETIOPAPI_API macro manually since */
-/* part of it must be toghether with the pointer name in the parentheses.        */
+/* part of it must be together with the pointer name in the parentheses.        */
 typedef NETIO_STATUS (NETIOAPI_API_ *pGetIfEntry2_t)(PMIB_IF_ROW2 Row);
 
 /* GetIfEntry2() is available since Windows Vista and Windows Server 2008. In    */
 /* earlier Windows releases this pointer remains set to NULL and GetIfEntry() is */
-/* used dirrectly instead.                                                       */
+/* used directly instead.                                                       */
```
Comment (jsoref) [inline src/libs/zbxsysinfo/win32/net.c]: I tried to fix all of the padding for `*/`, but it looks like i missed a few :-(

File: src/zabbix_server/vmware/vmware.h [hunk 13]
```diff
@@ -52,7 +52,7 @@ typedef struct
 	/*    pair->second - value                  */
 	zbx_vector_ptr_pair_t	values;
 
-	/* the counter state, see ZBX_VMAWRE_COUNTER_* defines */
+	/* the counter state, see ZBX_VMWARE_COUNTER_* defines */
```
Comment (jsoref) [inline src/zabbix_server/vmware/vmware.h]: this is a brand typo

File: ChangeLog [hunk 14]
```diff
@@ -9729,7 +9729,7 @@ Changes for 1.0alpha10:
  - fixed double appearing of complex triggers in list of triggers
  - added form for changing host details
  - All calculations are made in C code instead of SQL statements
- - Housekeping now runs as separate process, however it's part of zabix_suckerd
+ - Housekeeping now runs as separate process, however it's part of zabbix_suckerd
```
Comment (jsoref) [inline ChangeLog]: this is your brand...

**Comment by richlv** [comment]
it is really great that you spotted all these things :)
one thing that makes this harder to work on - a mix of spelling, formatting and variant changes. for example, the default locale is en_GB, thus the changes should have been the other way around (except that item helper strings are synced straight from the manual, so the issue is a bit more complicated...)
as for the older changelog entries, i have been intentionally leaving them as is, although "housekeping" could be changed anyway ;)

note that my feedback is based on my personal views, this is not affecting the merge-ability.

**Comment by jsoref** [comment]
Tentatively, I'll drop the following commits (probably tonight):
* spelling: customized
* spelling: optimization
* spelling: synchronization
* spelling: synchronizer
* spelling: uninitialize

The reason I leave individual word corrections as distinct commits is that it makes responding to feedback like this fairly easy. 

You should be able to review the list of commits and identify any words for which I've incorrectly picked a spelling. Please let me know if there are others which are problematic. (knowledge* can be)

**Comment by jsoref** [comment]
@richlv: I've excluded the en-US bits from the PR. If this is enough for people to move forward, that's great.

Otherwise, I'll need more guidance about what to do. I can certainly split things by category, but it takes time / effort.

e.g., I'd rather spend it getting zabbix running on one of my machines (I have it running on over a dozen, but one of them is being stubborn...).

**Comment by richlv** [comment]
i'd suggest creating a bugreport at https://support.zabbix.com . while normally it is suggested to attach patches to the issue there, i think in this case it would be ok to link to the pull request here instead :)

**Comment by jsoref** [comment]
@richlv: https://support.zabbix.com/browse/ZBX-11620

**Comment by richlv** [comment]
great, thank you :)

**Pull request closed without merging**